src/rpc/RPCServer.ts

```diff
@@ -253,7 +282,7 @@ class RPCServer {
       ctx,
     ) {
       for await (const inputVal of input) {
-        yield handler(inputVal, connectionInfo, ctx);
+        yield await handler(inputVal, connectionInfo, ctx);
         break;
       }
```

Comment:
Given that this is a unary handler, why are we doing a `for await` on the `input`?

Shouldn't we expect just one piece of data from the `input`?

That is instead of doing `for await`, it would be clearer to do:

```
const inputVal = input.next();
```

---

Oh wait, because it is an `AsyncIterable`, you have to use `for await`.

I am adding a comment here.

Reply:
It is possible to do this:

```
const inputVal = (await input[Symbol.asyncIterator]().next()).value;
```

However it's a bit wordy.